Formula/corectl.rb

```diff
@@ -13,27 +22,49 @@ class Corectl < Formula
 
   depends_on "go" => :build
   depends_on "godep" => :build
+  depends_on "ocaml" => :build
+  depends_on "opam" => :build
   depends_on :macos => :yosemite
 
   def install
     ENV["GOPATH"] = buildpath
 
+    opamroot = buildpath/"opamroot"
+    opamroot.mkpath
+    ENV["OPAMROOT"] = opamroot
+    ENV["OPAMYES"] = "1"
+
     path = buildpath/"src/github.com/TheNewNormal/#{name}"
     path.install Dir["*"]
 
     args = []
     args << "VERSION=#{version}" if build.stable?
 
     cd path do
-      system "make", "corectl", *args
-      system "make", "documentation/man"
-      bin.install "corectl"
+      system "opam", "init", "--no-setup"
+      system "opam", "pin", "add", "qcow-format",
+        "https://github.com/mirage/ocaml-qcow.git#master"
```

Comment:
@AntonioMeireles what you'll need to do is pin it to a specific #commit `if build.stable?`, and since you have a head url at the top too, use #master `if build.head?`